### PR TITLE
support template golden files

### PR DESCRIPTION
**Pull request by almalki**

### CODE REVIEW

**Comment by sebdah** [comment]
This is good stuff @almalki. I'll review it in a bit. Could you please add a note in the `README.md` about this feature?

File: goldie.go [hunk 1]
```diff
@@ -103,12 +116,24 @@ func compare(name string, actualData []byte) error {
...
+		return fmt.Errorf("Expected %s to be nil", err.Error())
+	}
+
+	var expectedData bytes.Buffer
+	err = tmpl.Execute(&expectedData, data)
+	if err != nil {
+		return fmt.Errorf("Expected %s to be nil", err.Error())
+	}
+
+
```
Comment (sebdah) [inline goldie.go]: Remove extra blank line please.

File: goldie_test.go [hunk 2]
```diff
@@ -165,7 +165,64 @@ func TestCompare(t *testing.T) {
...
-		err := compare(test.name, test.actualData)
+		err := compare(test.name, nil, test.actualData)
+		assert.IsType(t, test.err, err)
+
+		err = os.RemoveAll(FixtureDir)
+		assert.Nil(t, err)
+	}
+}
+
+
```
Comment (sebdah) [inline goldie_test.go]: Remove extra blank line please.

File: goldie_test.go [hunk 3]
```diff
@@ -165,7 +165,64 @@ func TestCompare(t *testing.T) {
...
+func TestCompareTemplate(t *testing.T) {
+
```
Comment (sebdah) [inline goldie_test.go]: Remove leading blank line please.

File: goldie.go [hunk 4]
```diff
@@ -72,6 +73,17 @@ func Assert(t *testing.T, name string, actualData []byte) {
...
+// golden files after executing it as a template with data parameter.
+// If the update flag is set, it will also update the golden file.
+// `name` refers to the name of the test and it should typically be unique
+// within the package. Also it should be a valid file name (so keeping to
+// `a-z0-9\-\_` is a good idea).
+func AssertWithTemplate(t *testing.T, name string, data interface{} , actualData []byte) {
+	Assert(t, name, nil, actualData)
+}
+
+
```
Comment (sebdah) [inline goldie.go]: Remove extra blank line please.

File: goldie.go [hunk 5]
```diff
@@ -49,7 +50,7 @@ var (
 // `name` refers to the name of the test and it should typically be unique
 // within the package. Also it should be a valid file name (so keeping to
 // `a-z0-9\-\_` is a good idea).
-func Assert(t *testing.T, name string, actualData []byte) {
+func Assert(t *testing.T, name string, data interface{}, actualData []byte) {
```
Comment (sebdah) [inline goldie.go]: This is a breaking change that we need to find a way around. A suggestion would be to skip the `data interface{}` parameter here and pass `nil` to `compare` always. What do you think?

File: goldie.go [hunk 6]
```diff
@@ -72,6 +83,7 @@ func Assert(t *testing.T, name string, actualData []byte) {
 	}
 }
 
+
```
Comment (sebdah) [inline goldie.go]: Remove extra blank line please.

File: goldie.go (hunk 5 above)
Comment (almalki) [reply goldie.go]: I fixed it, Assert will stay as it is, another func AssertWithTemplate will accept data

**Comment by sebdah** [comment]
Great stuff, thank you for this contribution. I'll merge it! :tada: :bowing_man:

**Pull request merged**